torch/_inductor/codegen/triton.py

```diff
@@ -1357,6 +1360,60 @@ def reduction_hint(node):
         else:
             return node.node.data.reduction_hint
 
+    @staticmethod
+    def estimate_reg_usage(list_nodes: List["SchedulerNode"]):
+        # We estimate the registry usage of a kernel by counting the number of alive variables
+        # in the graph at any given point and taking the max
+        # N.B. A tighter bound may be given by implementing algorithm like that proposed in
+        # Minimum register instruction sequence problem: revisiting optimal code generation for DAGs
+        # If you intend to do that, talk to Lezcano, as he knows how to improve the algorithm in that paper
+
+        die_now = []
+        dead_variables = set()
+        for n in reversed(list_nodes):
+            if n in (EnableReduction, DisableReduction):
+                die_now.append(set())
+            else:
+                # Use the writes as this may run before DCE
+                used_vars = set(itertools.chain((dep.name for dep in n.read_writes.reads),
+                                                (dep.name for dep in n.read_writes.writes)))
```

Comment:
This assumes one memory dep per buffer which seems wrong to me.

Reply:
Are you are saying that I shouldn't discard the index from the reads and writes? If that's the case, yep, I think you're right. I'll fix that.